presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -0,0 +1,703 @@
+        }
+
+        if (database.getLocation().isPresent()) {
+            HiveWriteUtils.createDirectory(hdfsContext, hdfsEnvironment, new Path(database.getLocation().get()));
+        }
+    }
+
+    @Override
+    public void dropDatabase(String databaseName)
+    {
+        try {
+            glueClient.deleteDatabase(new DeleteDatabaseRequest().withName(databaseName));
+        }
+        catch (EntityNotFoundException e) {
+            throw new SchemaNotFoundException(databaseName);
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public void renameDatabase(String databaseName, String newDatabaseName)
+    {
+        try {
+            Database db = getDatabase(databaseName).orElseThrow(() -> new SchemaNotFoundException(databaseName));
+            DatabaseInput renamedDb = GlueInputConverter.convertDatabase(db).withName(newDatabaseName);
+            glueClient.updateDatabase(new UpdateDatabaseRequest()
+                    .withName(databaseName)
+                    .withDatabaseInput(renamedDb));
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public void createTable(Table table, PrincipalPrivileges principalPrivileges)
+    {
+        try {
+            TableInput input = GlueInputConverter.convertTable(table);
+            glueClient.createTable(new CreateTableRequest()
+                    .withDatabaseName(table.getDatabaseName())
+                    .withTableInput(input));
+        }
+        catch (AlreadyExistsException e) {
+            throw new TableAlreadyExistsException(new SchemaTableName(table.getDatabaseName(), table.getTableName()));
+        }
+        catch (EntityNotFoundException e) {
+            throw new SchemaNotFoundException(table.getDatabaseName());
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public void dropTable(String databaseName, String tableName, boolean deleteData)
+    {
+        Table tbl = getTableOrElseThrow(databaseName, tableName);
+
+        try {
+            glueClient.deleteTable(new DeleteTableRequest()
+                    .withDatabaseName(databaseName)
+                    .withName(tableName));
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+
+        String tblLocation = tbl.getStorage().getLocation();
+        if (deleteData && !isExternalTable(tbl) && !isNullOrEmpty(tblLocation)) {
+            deleteDir(hdfsContext, hdfsEnvironment, new Path(tblLocation), true);
+        }
+    }
+
+    private static boolean isExternalTable(Table tbl)
+    {
+        return tbl.getTableType().equals(EXTERNAL_TABLE.name());
```

Comment:
We might want to make this `isManagedTable()` and check for `MANAGED_TABLE` explicitly. Otherwise, this could match other types like `VIRTUAL_VIEW`.

Reply:
Good catch, I'll check for managed table instead